<html> 
 <head> 
  <style> table, th, td {
  border: 1px solid black;
  border-collapse: collapse;
}th {
  background: lightblue;
}</style> 
 </head> 
 <body> 
  <h1>SATD</h1> 
  <table> 
   <tbody> 
    <tr> 
     <th>satd id</th> 
     <th>satd instance id</th> 
     <th>project</th> 
     <th>committer name </th> 
     <th> Commit Hash</th> 
     <th>old comment</th> 
     <th>New Comment</th> 
     <th>resolution</th> 
     <th>Method Signature</th> 
     <th>Method Declaration</th> 
     <th>Method Body</th> 
    </tr> 
    <tr> 
     <td>3895</td> 
     <td>-1422651683</td> 
     <td>apache/camel</td> 
     <td>Claus Ibsen</td> 
     <td>69c0182ef98463324f41dd6f59b2647d90c1e00b</td> 
     <td>TODO set the corralationID</td> 
     <td>TODO set the correlationID</td> 
     <td>SATD_CHANGED</td> 
     <td>process(Exchange)</td> 
     <td>public void process(final Exchange exchange) throws Exception</td> 
     <td> org.springframework.messaging.Message<!--?--> request = SpringIntegrationBinding.createSpringIntegrationMessage(exchange); if (exchange.getPattern().isOutCapable()) { exchange.getIn().getHeaders().put(MessageHeaders.REPLY_CHANNEL, replyChannel); // we want to do in-out so the inputChannel is mandatory (used to receive reply from spring integration) if (replyChannel == null) { throw new IllegalArgumentException("ReplyChannel has not been configured on: " + this); } replyChannel.subscribe(new MessageHandler() { public void handleMessage(Message<!--?--> message) { LOG.debug("Received {} from ReplyChannel: {}", message, replyChannel); // TODO set the correlationID SpringIntegrationBinding.storeToCamelMessage(message, exchange.getOut()); } }); } requestChannel.send(request); </td> 
    </tr> 
   </tbody> 
   <tbody> 
    <tr> 
     <td>1321</td> 
     <td>-1422651683</td> 
     <td>apache/camel</td> 
     <td>Claus Ibsen</td> 
     <td>35ac5413ec820d26624598f48b9a4d22394f4415</td> 
     <td>TODO set the corralationID</td> 
     <td>TODO set the corralationID</td> 
     <td>CLASS_OR_METHOD_CHANGED</td> 
     <td>process(Exchange)</td> 
     <td>public void process(final Exchange exchange) throws Exception</td> 
     <td> org.springframework.integration.core.Message request = SpringIntegrationBinding.createSpringIntegrationMessage(exchange); if (exchange.getPattern().isOutCapable()) { exchange.getIn().getHeaders().put(MessageHeaders.REPLY_CHANNEL, replyChannel); replyChannel.subscribe(new MessageHandler() { public void handleMessage(Message<!--?--> message) { // TODO set the corralationID SpringIntegrationBinding.storeToCamelMessage(message, exchange.getOut()); } }); } requestChannel.send(request); </td> 
    </tr> 
   </tbody> 
   <tbody> 
    <tr> 
     <td>864</td> 
     <td>-1422651683</td> 
     <td>apache/camel</td> 
     <td>Willem Ning Jiang</td> 
     <td>073a062146e15a3c21c66e3886ea28d10dc49be4</td> 
     <td>TODO Maybe we should set the exception as the fault message</td> 
     <td>TODO set the corralationID</td> 
     <td>SATD_CHANGED</td> 
     <td>process(Exchange, AsyncCallback)</td> 
     <td>public boolean process(final Exchange exchange, final AsyncCallback callback)</td> 
     <td> org.springframework.integration.core.Message request = SpringIntegrationBinding.createSpringIntegrationMessage(exchange); Map<string, object>
        headers = new HashMap 
       <string, object>
         (); if (exchange.getPattern().isOutCapable()) { headers.put(MessageHeaders.REPLY_CHANNEL, replyChannel); replyChannel.subscribe(new MessageHandler() { public void handleMessage(Message<!--?--> message) { // TODO set the corralationID SpringIntegrationBinding.storeToCamelMessage(message, exchange.getOut()); callback.done(true); } }); } requestChannel.send(request); if (!exchange.getPattern().isOutCapable()) { callback.done(true); } return true; 
       </string,> 
      </string,></td> 
    </tr> 
   </tbody> 
   <tbody>
    <tr>
     <td>649</td> 
     <td>-1422651683</td>
     <td>apache/camel</td>
     <td>James Strachan</td>
     <td>2191af857dc2238533d88d0bc1519e1c2c6fd5c3</td> 
     <td>None</td> 
     <td>TODO Maybe we should set the exception as the fault message</td> 
     <td>SATD_ADDED</td> 
     <td>process(Exchange)</td> 
     <td>public void process(Exchange exchange)</td> 
     <td> try { incoming(exchange); } catch (Throwable ex) { ex.printStackTrace(); logger.warn("Failed to process incoming message : " + ex); // TODO Maybe we should set the exception as the fault message } </td> 
    </tr>
   </tbody>
  </table>  
 </body>
</html>